src/Sylius/Bundle/ApiBundle/spec/Validator/CheckoutCompletion/CartStateValidatorSpec.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\CheckoutCompletion;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachineInterface;
+use Sylius\Bundle\ApiBundle\Command\OrderTokenValueAwareInterface;
+use Sylius\Bundle\ApiBundle\Validator\CheckoutCompletion\CartState;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\OrderCheckoutStates;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+final class CartStateValidatorSpec extends ObjectBehavior
+{
+    function let(
+        OrderRepositoryInterface $orderRepository,
+        FactoryInterface $stateMachineFactory,
+        RouterInterface $router,
+    ): void {
+        $this->beConstructedWith($orderRepository, $stateMachineFactory, $router);
+    }
+
+    function it_is_a_constraint_validator(): void
+    {
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_if_value_is_not_an_instance_of_order_token_value_aware_interface(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', ['', new class() extends Constraint {
+            }])
+        ;
+    }
+
+    function it_throws_an_exception_if_constraint_is_not_an_instance_of_cart_state(
+        OrderTokenValueAwareInterface $orderTokenValueAware,
+    ): void {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', [$orderTokenValueAware, new class() extends Constraint {
```

Comment:
```suggestion
            ->during('validate', [$orderTokenValueAware, $constraint])
```
You can just mock it